### PR TITLE
Sync Ceph working branch part 2

### PR DESCRIPTION
**Pull request by xarses**
This pull request contains updates to the ceph pupet module and some (disabled) hooks into cluster_ha and cluster_simple.

Can now deploy ceph-osd roles if using https://github.com/xarses/fuelweb/tree/ceph-stage-3 and changing use_ceph to true in osnailyfacter/manafests/cluster_ha.pp or cluster_simple.pp

What works:
- ceph-mon's are installed on the controllers.
- In cluster_ha, monitors are joined sequentially
- ceph-osd's are created for each device provisioned from the fuelweb ceph-stage-3 branch.
- ceph cluster comes online and active+clean when two or more ceph-osd roles are defined

What's broken:
- ceph::glance doesn't run for some reason but glance::backend::ceph does, so there are missing parts (ceph.client.images.keyring)
- ceph::cinder doesn't run in cluster_simple but usually does in cluster_ha
- ceph-osd role doesn't run on primary_controller using cluster_ha
- ceph_conf ini helper doesn't replace existing values, it adds additional statements. ceph is smart, and only complains.

### CODE REVIEW

File: deployment/puppet/ceph/lib/facter/ceph_osd.rb [hunk 1]
```diff
@@ -1,6 +1,17 @@
 Facter.add("osd_devices_list") do
     setcode do
         # Use any filesystem labeled "cephosd" as an osd
-        %x{blkid -o list | awk '{if ($3 == "cephosd") print $1}'}
+        devs = %x{blkid -o list | awk '{if ($3 == "cephosd") print $1}'}.split("\n")
+        journal = %x{blkid -o list | awk '{if ($3 == "ceph-journal") print $4}'}.strip
```
Comment (xdel) [inline deployment/puppet/ceph/lib/facter/ceph_osd.rb]: may I suggest to remove dash here?

File: deployment/puppet/ceph/manifests/deploy.pp [hunk 2]
```diff
@@ -1,129 +1,6 @@
+#Ceph::deploy will installl mds server if invoked
```
Comment (xdel) [inline deployment/puppet/ceph/manifests/deploy.pp]: typo

File: deployment/puppet/ceph/manifests/keystone.pp [hunk 3]
```diff
@@ -17,12 +18,12 @@
       | certutil -d ${directory} -A -n ca -t 'TCu,Cu,Tuw' && openssl x509  \
       -in /etc/keystone/ssl/certs/signing_cert.pem -pubkey | certutil -A -d \
       ${directory} -n signing_cert -t 'P,P,P'",
-      require => [File["${directory}"], Package["libnss3-tools"]]
+      require => [File["${directory}"], Package['libnss3-tools']]
```
Comment (xdel) [inline deployment/puppet/ceph/manifests/keystone.pp:21]: please add if for ubuntu and rh

File: deployment/puppet/ceph/manifests/keystone.pp [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
-  $adm_ip,
-  $int_ip,
+  $pub_ip = $::ceph::rgw_pub_ip,
+  $adm_ip = $::ceph::rgw_adm_ip,
+  $int_ip = $::ceph::rgw_int_ip,
   $directory = '/etc/ceph/nss',
 ) {
   if str2bool($::keystone_conf) {
-    package { "libnss3-tools" :
+    package { 'libnss3-tools' :
```
Comment (xdel) [inline deployment/puppet/ceph/manifests/keystone.pp:9]: please add if for ubuntu and rh

File: deployment/puppet/ceph/manifests/keystone.pp (hunk 3 above)
Comment (xarses) [inline deployment/puppet/ceph/manifests/keystone.pp:21]: keystone.pp is for radios support which still in progress

**Pull request merged**